### PR TITLE
Branch1

### PR DESCRIPTION
**Pull request by qitshen-sh**
add test

### CODE REVIEW

File: .idea/libraries/Maven__commons_codec_commons_codec_1_14.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<component name="libraryTable">
```
Comment (sutyan-sh) [inline .idea/libraries/Maven__commons_codec_commons_codec_1_14.xml:1]: files under .idea no need to commit

Comment (sutyan-sh) [reply .idea/libraries/Maven__commons_codec_commons_codec_1_14.xml:1]: you can use .gitignore file to exclude this kind of file

File: demo.iml [hunk 2]
```diff
@@ -54,6 +54,22 @@
     <orderEntry type="library" name="Maven: org.springframework:spring-aop:5.2.9.RELEASE" level="project" />
     <orderEntry type="library" name="Maven: org.springframework:spring-context:5.2.9.RELEASE" level="project" />
     <orderEntry type="library" name="Maven: org.springframework:spring-expression:5.2.9.RELEASE" level="project" />
+    <orderEntry type="library" name="Maven: io.rest-assured:rest-assured:3.0.3" level="project" />
```
Comment (sutyan-sh) [inline demo.iml:57]: this file can be excluded as well

File: src/main/java/com/example/demo/MessageController.java [hunk 3]
```diff
@@ -40,10 +40,11 @@ public Message Greet(){
     @PostMapping("/message")
     public Message PostMessage(@RequestBody Message message){
```
Comment (sutyan-sh) [inline src/main/java/com/example/demo/MessageController.java:41]: should use ResponseEntity,  and for create record should return 201

File: src/test/java/com/example/demo/DemoApplicationTests.java [hunk 4]
```diff
@@ -1,13 +1,57 @@
...
+	private static final String BASE = "http://localhost:8080";
+
+	@Test
+	public void testGetGreeting(){
+		get(BASE + "/greet").then().contentType("application/json").body("message", equalTo("Greetings.")).statusCode(200);
+	}
+
+	@Test
+	public void testPostGreeting(){
+		post(BASE + "/greet").then().statusCode(405);
```
Comment (sutyan-sh) [inline src/test/java/com/example/demo/DemoApplicationTests.java:26]: 405 is bad response

Comment (sutyan-sh) [reply src/test/java/com/example/demo/DemoApplicationTests.java:26]: do you know when will throw 405?

File: .idea/libraries/Maven__commons_codec_commons_codec_1_14.xml (hunk 1 above)
Comment (qitshen-sh) [reply .idea/libraries/Maven__commons_codec_commons_codec_1_14.xml:1]: Thank you very much for your advice.

**Pull request merged**